packages/components/src/alignment-matrix-control/test/index.tsx

```diff
@@ -17,34 +17,115 @@ const getCell = ( name: string ) => {
 	return within( getControl() ).getByRole( 'gridcell', { name } );
 };
 
+const asyncRender = async ( jsx: any ) => {
+	const view = render( jsx );
+	await act( async () => {
+		// This allows the component to properly establish
+		// its initial state, that sometimes isn't otherwise
+		// ready in time for tests to start.
+		await new Promise( requestAnimationFrame );
+		await new Promise( requestAnimationFrame );
+		await new Promise( requestAnimationFrame );
+	} );
```

Comment:
The problem with the 3 `await new Promise( requestAnimationFrame );` is that they're prone to error.

Since this is just a way to add extra "waiting" before tests start, theoretically, it might happen that this added time is still not enough for us to be certain that the component has finished initializing its state.

As a more reliable alternative, we can look at how the component reacts to the state initialization. In other words, we can observe what in the component is being modified *after* the state initialization.

Luckily for us, there is one change that we can use: after the composite state initializes, each cell will get a `tabindex="-1"`, except the default one. Since `tabindex` by default is `0`, that means we can wait for it to change to `-1`, and that means state initialization has already finished.

```suggestion
	// This ensures the component has properly established its initial state.
	await waitFor( () => {
		expect( getCell( 'top left' ) ).toHaveAttribute( 'tabindex', '-1' );
	} );
```